File: Python/Merge Two Sorted Lists.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    # @param two ListNodes
    # @return a ListNode
    def mergeTwoLists(self, l1, l2):
        flag = 0
        while l1 != None and l2 != None:
            if l1.val < l2.val:
                if flag == 0:
                    head = l1
                    node = l1
                    flag = 1
                else:
                    node.next = l1
                    node = node.next
                l1 = l1.next
            else:
                if flag == 0:
                    head = l2
                    node = l2
                    flag = 1
                else:
                    node.next = l2
                    node = node.next
                l2 = l2.next
        if l1 == None:
            if flag == 0:
                head = l2
            else:
                node.next = l2
        else:
            if flag == 0:
                head = l1
            else:
                node.next = l1
        return head

File: C++/Merge Two Sorted Lists.cpp
#include<iostream>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
  };


ListNode *mergeTwoLists(ListNode *l1,ListNode *l2)
{
    if(l1==NULL) return l2;
    if(l2==NULL) return l1;
    ListNode *node;
    if(l1->val<l2->val)
    {
        node=l1->next;
        l1->next=mergeTwoLists(node, l2);
        return l1;
    }
    else
    {
        node=l2->next;
        l2->next=mergeTwoLists(l1, node);
        return l2;
    }
    return NULL;
}
